lms/services/blackboard_api/_basic.py

```diff
@@ -61,20 +62,29 @@ def __init__(
         self.redirect_uri = redirect_uri
 
         self._http_service = http_service
+        self._oauth_http_service = oauth_http_service
         self._oauth2_token_service = oauth2_token_service
 
-    def get_token(self, authorization_code):
+    def get_token(self, authorization_code=None, refresh_token=None):
```

Comment:
I would have expected to see this sort of functionality (or at least quite a bit of it) inside the `OAuthHTTPService`. I think we're on the right track when `oauth2_token_service` is not required in this class. That would show that `OAuthHTTPService` has a nice boundary and is doing an encapsulated task.

Is there a way we can supply the URL or any variable parts of this to a more general `get_token()` function in `OAuthHTTPService`?

Reply:
I don't think we should try to answer that question until we have two or three users of `OAuthHTTPService`. At that point it should be clear how to implement a generic `get_token()`

Reply:
I did end up moving `access_token_request()` and `refresh_token_request()` into a generic place in `OAuthHTTPService`: https://github.com/hypothesis/lms/pull/2987 and https://github.com/hypothesis/lms/pull/2988